core/ledger/kvledger/bookkeeping/provider.go

```diff
@@ -60,7 +62,10 @@ func (provider *provider) Drop(ledgerID string) error {
 	if err := provider.dbProvider.Drop(dbName(ledgerID, PvtdataExpiry)); err != nil {
 		return err
 	}
-	return provider.dbProvider.Drop(dbName(ledgerID, MetadataPresenceIndicator))
+	if err := provider.dbProvider.Drop(dbName(ledgerID, MetadataPresenceIndicator)); err != nil {
+		return err
+	}
+	return provider.dbProvider.Drop(dbName(ledgerID, SnapshotRequest))
```

Comment:
Earlier, it was two categories. Given that it has become three, I would prefer doing something like this

```
	for _, c := range []Category{PvtdataExpiry, MetadataPresenceIndicator, SnapshotRequest} {
		if err := provider.dbProvider.Drop(dbName(ledgerID, c)); err != nil {
			return err
		}
	}
	return nil

```